flowermd/modules/welding/welding.py

```diff
@@ -25,112 +26,150 @@ class Interface:
 
     """
 
-    def __init__(self, gsd_file, interface_axis, gap, wall_sigma=1.0):
-        self.gsd_file = gsd_file
+    def __init__(
+        self,
+        gsd_files,
+        interface_axis,
+        gap,
+        wall_sigma=1.0,
+        remove_void_particles=True,
+    ):
+        self.gsd_files = check_return_iterable(gsd_files)
         self.interface_axis = interface_axis
         self.gap = gap
         self.wall_sigma = wall_sigma
+        self._remove_void_particles = remove_void_particles
         self.hoomd_snapshot = self._build()
 
     def _build(self):
         """Duplicates the slab and builds the interface."""
-        gsd_file = gsd.hoomd.open(self.gsd_file)
-        snap = gsd_file[-1]
-        gsd_file.close()
-        axis_index = np.where(self.interface_axis != 0)[0]
+        if len(self.gsd_files) == 1:
+            gsd_file_L = gsd.hoomd.open(self.gsd_files[0])
+            gsd_file_R = gsd.hoomd.open(self.gsd_files[0])
+        else:
+            gsd_file_L = gsd.hoomd.open(self.gsd_files[0])
+            gsd_file_R = gsd.hoomd.open(self.gsd_files[1])
+        # Get snapshots
+        snap_L = gsd_file_L[-1]
+        snap_R = gsd_file_R[-1]
+        gsd_file_L.close()
+        gsd_file_R.close()
+        # Remove VOID partilces if needed
+        if self._remove_void_particles:
+            for snap in [snap_L, snap_R]:
+                if "VOID" in snap.particles.types:
+                    snap.particles.N -= 1
+                    void_id = snap.particles.types.index("VOID")
+                    snap.particles.types.remove("VOID")
+                    keep_indices = np.where(snap.particles.typeid != void_id)[0]
+                    snap.particles.position = snap.particles.position[
+                        keep_indices
+                    ]
+                    snap.particles.mass = snap.particles.mass[keep_indices]
+                    snap.particles.charge = snap.particles.charge[keep_indices]
+                    snap.particles.orientation = snap.particles.orientation[
+                        keep_indices
+                    ]
+                    snap.particles.typeid = snap.particles.typeid[keep_indices]
 
         interface = gsd.hoomd.Frame()
-        interface.particles.N = snap.particles.N * 2
-        interface.bonds.N = snap.bonds.N * 2
-        interface.bonds.M = snap.bonds.M
-        interface.angles.N = snap.angles.N * 2
-        interface.angles.M = snap.angles.M
-        interface.dihedrals.N = snap.dihedrals.N * 2
-        interface.dihedrals.M = snap.dihedrals.M
-        interface.pairs.N = snap.pairs.N * 2
+        interface.particles.N = snap_L.particles.N + snap_R.particles.N
+        interface.bonds.N = snap_L.bonds.N + snap_R.bonds.N
+        # TODO: Do we need bonds.M?
```

Comment:
The `M` values for these classes are always gonna be a [fixed number.](https://gsd.readthedocs.io/en/stable/python-module-gsd.hoomd.html#gsd.hoomd.BondData) M=2 for bond, M=3 for angle and M=4 for dihedrals.